src/app/@ansyn/ansyn/app-effects/effects/map.app.effects.ts

```diff
@@ -321,10 +334,40 @@ export class MapAppEffects {
 			})
 		);
 
+	@Effect({ dispatch: false })
+	searchByExtentPolygon$: Observable<any> = this.actions$.pipe(
+		ofType(MapActionTypes.POSITION_CHANGED, MapActionTypes.SET_ACTIVE_MAP_ID),
+		withLatestFrom(this.store$.select(selectMaps), this.store$.select(selectActiveMapId), this.store$.select(selectGeoFilterStatus)),
+		map(([action, mapList, activeMapId, geoFilterStatus]) => {
+			const activeMap: IMapSettings = mapList[activeMapId];
+			return [activeMap.data.position, geoFilterStatus];
+		}),
+		filter(([position, geoFilterStatus]: [IImageryMapPosition, IGeoFilterStatus]) => Boolean(position) && geoFilterStatus.type === CaseGeoFilter.ScreenView),
+		debounceTime(VIEW_SEARCH_DEBOUNCE_TIME),
+		tap(([position, geoFilterStatus]: [IImageryMapPosition, IGeoFilterStatus]) => {
+			let zoom = position.projectedState.zoom;
+
+			if (!equalPolygons(position.extentPolygon, extentPolygon)) {
+				if (zoom < VIEW_SEARCH_ZOOM_LIMIT) {
```

Comment:
won't work on overlays
you can't use zoom on images only resolution - try to use the scale - or calculate BBOX width size in meters